app/models/registry.rb

```diff
@@ -20,12 +22,47 @@ def global_namespace
     Namespace.find_by(registry: self, global: true)
   end
 
+  def client
+    RegistryClient.new(hostname, use_ssl)
+  end
+  memoize :client
```

Comment:
We are cautious when it comes to introducing new dependencies: it takes time and resources to keep them up-to-date and secure. 

In this case I would drop this gem and dome something simple like:

```ruby
def client
  @client = RegistryClient.new(hostname, use_ssl) unless @client
  @client   
end
```